src/src/PluginStructs/P148_data_struct.cpp

```diff
@@ -88,24 +88,23 @@ uint8_t P148_data_struct::TM1621GetFontCharacter(char character, bool firstrow)
   return 0u;
 }
 
+// Do not change the order of these as it is stored.
 union MonitorTaskValue_conversion {
   struct {
-    int16_t TaskIndex : 9;
-    int16_t taskVar   : 3;
-    int16_t unit      : 3;
-    int16_t showname  : 1;
+    uint16_t TaskIndex : 9;
+    uint16_t unit      : 3;
+    uint16_t taskVar   : 3;
+    uint16_t showname  : 1;
```

Comment:
Should these default values have the `u` postfix, now that the types have changed to unsigned?

Reply:
These are only used for making the conversion.
I used the 'tricky' union type to pack and unpack bits.
As you can see, I also reserved 1 extra bit for the taskvar (and one for the taskindex, which is a bit presumptuous I now realize)
I needed to use uint types here as C++ would otherwise still keep the "signed" bit and then you effectively loose 1 bit when storing values in here.

I will make one last change here, since nobody uses it yet, to reduce the taskindex nr. of bits and pad it with some dummy bits for future use.
But the important take-away from this is that you need to use uint types when packing bits like this.
This took me about 2+ hours yesterday to find out why the set "unit" wasn't working well.

Reply:
Ah, my eyes translated the `:` to `=` 😊, need (more) coffee...